safe_transaction_service/history/tests/test_views_v2.py

```diff
@@ -145,3 +156,316 @@ def test_safe_collectibles_paginated(self):
+            "safe_transaction_service.history.serializers.get_safe_owners",
+            return_value=[Account.create().address],
+        ) as get_safe_owners_mock:
+            response = self.client.post(url, format="json", data=data)
+            self.assertIn(
+                f"Provided delegator={delegator.address} is not an owner of Safe={safe_address}",
+                response.data["non_field_errors"][0],
+            )
+            self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+            get_safe_owners_mock.return_value = [delegator.address]
+            response = self.client.post(url, format="json", data=data)
+            self.assertIn(
+                f"Signature does not match provided delegator={delegator.address}",
+                response.data["non_field_errors"][0],
+            )
+            self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+            # Create delegate
+            self.assertEqual(SafeContractDelegate.objects.count(), 0)
+            chain_id = self.ethereum_client.get_chain_id()
+            hash_to_sign = DelegateSignatureHelperV2.calculate_hash(
+                delegate.address, chain_id, False
+            )
+            data["signature"] = delegator.signHash(hash_to_sign)["signature"].hex()
+            response = self.client.post(url, format="json", data=data)
+            self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+            safe_contract_delegate = SafeContractDelegate.objects.get()
+            self.assertEqual(safe_contract_delegate.delegate, delegate.address)
+            self.assertEqual(safe_contract_delegate.delegator, delegator.address)
+            self.assertEqual(safe_contract_delegate.label, label)
+            self.assertEqual(safe_contract_delegate.safe_contract_id, safe_address)
+
+            # Update label
+            label = "Jimmy McGill"
+            data["label"] = label
+            response = self.client.post(url, format="json", data=data)
+            self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+            self.assertEqual(SafeContractDelegate.objects.count(), 1)
+            safe_contract_delegate = SafeContractDelegate.objects.get()
+            self.assertEqual(safe_contract_delegate.label, label)
+
+        # Create delegate without a Safe
+        hash_to_sign = DelegateSignatureHelperV2.calculate_hash(
+            delegate.address, chain_id, False
+        )
+        data = {
+            "label": "Kim Wexler",
+            "delegate": delegate.address,
+            "delegator": delegator.address,
+            "signature": delegator.signHash(hash_to_sign)["signature"].hex(),
+        }
+        response = self.client.post(url, format="json", data=data)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(SafeContractDelegate.objects.count(), 2)
+
+        # Test not internal server error on contract signature
```

Comment:
What means this test? that we don't return a 500 for a specific case?

Reply:
I have just updated the commentary to avoid confusion.